snuba/reader.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+
+import re
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import TYPE_CHECKING, Any, Mapping, Optional, Sequence
+
+from dateutil.tz import tz
+
+
+if TYPE_CHECKING:
+    from mypy_extensions import TypedDict
+
+    Column = TypedDict("Column", {"name": str, "type": str})
+    Result = TypedDict(
```

Comment:
Sorry, let me rephrase:
It is not immutability that affects multiple implementations. Immutability is good per se. Dataclasses may make it easier to manage multiple implementation since we can have inheritance (I don't htink we can extend a Typed dictionary)

Reply:
Ah, yeah — I agree with this.